coverage.py

```diff
@@ -469,21 +555,76 @@ def set_presentation_ready(self, identifier):
         work.set_presentation_ready()
         return identifier
 
-    def add_coverage_record_for(self, identifier):
+    #
+    # Implementation of BaseCoverageProvider virtual methods.
+    #
+
+    def items_that_need_coverage(self, identifiers=None):
+        """Find all items lacking coverage from this CoverageProvider.
+
+        Items should be Identifiers, though Editions should also work.
+
+        By default, all identifiers of the `input_identifier_types` which
+        don't already have coverage are chosen.
+        """
+        qu = Identifier.missing_coverage_from(
+            self._db, self.input_identifier_types, self.output_source,
+            count_as_missing_before=self.cutoff_time, operation=self.operation
+        )
+        if identifiers:
+            qu = qu.filter(Identifier.id.in_([x.id for x in identifiers]))
+        return qu
+
+    def add_coverage_record_for(self, item):
+        """Record this CoverageProvider's coverage for the given
+        Edition/Identifier, as a CoverageRecord.
+        """
         return CoverageRecord.add_for(
-            identifier, self.output_source, operation=self.operation
+            item, data_source=self.output_source, operation=self.operation
         )
 
-    def process_item(self, identifier):
-        raise NotImplementedError()
+    def record_failure_as_coverage_record(self, failure):
+        """Turn a CoverageFailure into a CoverageRecord object."""
+        return failure.to_coverage_record(operation=self.operation)
 
-    def finalize_batch(self):
-        """Do whatever is necessary to complete this batch before moving on to
-        the next one.
-        
-        e.g. uploading a bunch of assets to S3.
+class WorkCoverageProvider(BaseCoverageProvider):
+
+    #
+    # Implementation of BaseCoverageProvider virtual methods.
+    #
+
+    def items_that_need_coverage(self, identifiers=None):
+        """Find all Works lacking coverage from this CoverageProvider.
+
+        By default, all Works which don't already have coverage are
+        chosen.
+
+        :param: Only Works connected with one of the given identifiers
+        are chosen.
         """
-        pass
+        qu = Work.missing_coverage_from(
+            self._db, operation=self.operation, 
+            count_as_missing_before=self.cutoff_time
+        )
+        if identifiers:
```

Comment:
If no identifiers are provided we don't need to do the join against LicensePool at all.